src/Form/MultiLinkField.php

```diff
@@ -0,0 +1,171 @@
+            $this->loadFrom($data);
+            return $this;
+        }
+
+        $ids = $this->cooerceValueToArray($value);
+        return parent::setValue($ids, $data);
+    }
+
+    public function saveInto(DataObjectInterface $record)
+    {
+        $fieldName = $this->getName();
+        if (!$fieldName) {
+            throw new LogicException('LinkField must have a name');
+        }
+
+        $relation = $record->hasMethod($fieldName) ? $record->$fieldName() : null;
+        if (!$relation) {
+            throw new LogicException("{$record->ClassName} is missing the relation '$fieldName'");
+        }
+
+        // Use RelationList rather than Relation here since some Relation classes don't allow setting value - but RelationList does.
+        if (!($relation instanceof RelationList || $relation instanceof UnsavedRelationList)) {
+            throw new LogicException("'$fieldName()' method on {$record->ClassName} doesn't return a relation list");
+        } else {
+            $relation->setByIDList($this->getValueArray() ?? []);
+        }
+
+        return $this;
+    }
+
+    public function getSchemaDataDefaults()
+    {
+        $data = parent::getSchemaDataDefaults();
+        $data['isMulti'] = true;
+        return $data;
+    }
+
+    public function getSchemaStateDefaults()
+    {
+        $data = parent::getSchemaStateDefaults();
+        $data['value'] = $this->getValueArray();
+        return $data;
+    }
+
+    protected function getDefaultAttributes(): array
+    {
+        $attributes = parent::getDefaultAttributes();
+        $attributes['data-value'] = $this->getValueArray();
+        return $attributes;
+    }
+
+    /**
+     * Extracts the value of this field, normalised as a non-associative array.
+     */
+    private function getValueArray(): array
+    {
+        return $this->cooerceValueToArray($this->Value());
+    }
+
+    /**
+     * Cooerces the value to an array if possible.
+     * @throws LogicException if the type cannot be cooerced into an array.
+     */
+    private function cooerceValueToArray(mixed $value): array
+    {
+        // Prepare string by removing whitespace from the ends
+        // A comma separated list of IDs will be turned into an array of IDs
+        // Anything else will either get caught in the empty check or the !is_iterable check
+        if (is_string($value)) {
+            $value = $this->coeerceCommaSeparatedString(trim($value));
+        }
+
+        if (empty($value)) {
+            return [];
+        }
+
+        if ($value instanceof SS_List) {
+            return $value->column('ID');
+        }
+
+        if (!is_iterable($value)) {
+            return [$value];
+        }
+
+        if (is_iterable($value) && !is_array($value)) {
+            return [...$value];
+        }
+
+        if (is_array($value)) {
+            return array_values($value);
+        }
+
+        throw new LogicException('Unexpected value type ' . gettype($value));
```

Comment:
```suggestion
        if (is_string($value)) {
            $value = $this->convertCommaSeparatedString(trim($value));
        }
        if (empty($value)) {
            return [];
        }
        if ($value instanceof SS_List) {
            return $value->column('ID');
        }
        if (!is_iterable($value)) {
            return [$value];
        }
        if (is_iterable($value) && !is_array($value)) {
            return [...$value];
        }
        if (is_array($value)) {
            return array_values($value);
        }
        throw new LogicException('Unexpected value type ' . gettype($value));
```
All the blank lines aren't needed since there's closing brackets to visually split everything up

Reply:
Need to unit test this

Reply:
> All the blank lines aren't needed since there's closing brackets to visually split everything up

We don't have any standards around that - that's entirely subjective and comes up time and again where one person wants new lines and another person doesn't. I don't think we should bother pulling each other up on these sorts of changes where we don't actually have any rules about how it should be done.

In this case I've made the change to avoid ping pong, but in the future I might push back because it really shouldn't be mentioned in the first place unless/until we have rules about it.

Reply:
Added unit test